### PR TITLE
EE: Do not trim comments from expression preceding a quote

### PR DESCRIPTION
**Pull request by tmat**
Workaround to address https://dev.azure.com/devdiv/DevDiv/_workitems/edit/847849.

The real fix would be https://github.com/dotnet/roslyn/issues/37536 but it is much more involved.

### CODE REVIEW

**Comment by tmat** [comment]
@ivanbasov @cston @jinujoseph PTAL

File: src/ExpressionEvaluator/CSharp/Source/ResultProvider/CSharpFormatter.cs [hunk 1]
```diff
@@ -34,6 +34,8 @@ internal override bool IsWhitespace(char c)
             return SyntaxFacts.IsWhitespace(c);
         }
 
+        // TODO: https://github.com/dotnet/roslyn/issues/37536 
+        // This parsing is imprecise and amy result in bad expressions.
```
Comment (cston) [inline src/ExpressionEvaluator/CSharp/Source/ResultProvider/CSharpFormatter.cs]: >amy [](start = 41, length = 3)

Typo

File: src/ExpressionEvaluator/VisualBasic/Source/ResultProvider/VisualBasicFormatter.vb [hunk 2]
```diff
@@ -39,14 +39,21 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.ExpressionEvaluator
             Return SyntaxFacts.IsWhitespace(c)
         End Function
 
+        ' TODO https://github.com/dotnet/roslyn/issues/37536 
+        ' This parsing Is imprecise And amy result in bad expressions.
```
Comment (cston) [inline src/ExpressionEvaluator/VisualBasic/Source/ResultProvider/VisualBasicFormatter.vb]: >Is imprecise And [](start = 23, length = 16)

Lowercase

Comment (cston) [inline src/ExpressionEvaluator/VisualBasic/Source/ResultProvider/VisualBasicFormatter.vb]: >amy [](start = 40, length = 3)

Typo

**Pull request merged**